freqtrade/data/metrics.py

```diff
@@ -200,28 +200,54 @@ def calculate_expectancy(trades: pd.DataFrame) -> float:
     :param trades: DataFrame containing trades (requires columns close_date and profit_abs)
     :return: expectancy
     """
+
     if len(trades) == 0:
         return 0
 
-    expectancy = 1
-
-    profit_sum = trades.loc[trades['profit_abs'] > 0, 'profit_abs'].sum()
-    loss_sum = abs(trades.loc[trades['profit_abs'] < 0, 'profit_abs'].sum())
-    nb_win_trades = len(trades.loc[trades['profit_abs'] > 0])
-    nb_loss_trades = len(trades.loc[trades['profit_abs'] < 0])
+    winning_trades = trades.loc[trades['profit_abs'] > 0]
+    losing_trades = trades.loc[trades['profit_abs'] < 0]
+    profit_sum = winning_trades['profit_abs'].sum()
+    loss_sum = abs(losing_trades['profit_abs'].sum())
+    nb_win_trades = len(winning_trades)
+    nb_loss_trades = len(losing_trades)
 
-    if (nb_win_trades > 0) and (nb_loss_trades > 0):
-        average_win = profit_sum / nb_win_trades
-        average_loss = loss_sum / nb_loss_trades
-        risk_reward_ratio = average_win / average_loss
-        winrate = nb_win_trades / len(trades)
-        expectancy = ((1 + risk_reward_ratio) * winrate) - 1
-    elif nb_win_trades == 0:
-        expectancy = 0
+    average_win = (profit_sum / nb_win_trades) if nb_win_trades > 0 else 0
+    average_loss = (loss_sum / nb_loss_trades) if nb_loss_trades > 0 else 0
+    winrate = (nb_win_trades / len(trades))
+    loserate = (nb_loss_trades / len(trades))
+    expectancy = (winrate * average_win) - (loserate * average_loss)
 
     return expectancy
 
 
+def calculate_expectancy_ratio(trades: pd.DataFrame) -> float:
```

Comment:
Can you help me understand this?

the function that used to be `calculate_expectancy` is now `calculate_expectancy_ratio` - and the old one is now returning the actual expectancy (not the ratio), right?